samples/Azure.ResourceManager.Sample/Generated/Models/VirtualMachineScaleSetListOSUpgradeHistory.Serialization.cs

```diff
@@ -38,7 +38,7 @@ internal static VirtualMachineScaleSetListOSUpgradeHistory DeserializeVirtualMac
                         property.ThrowNonNullablePropertyIsNull();
                         continue;
                     }
-                    etag = new ETag(property.Value.GetString());
+                    etag = JsonSerializer.Deserialize<ETag>(property.Value.ToString());
```

Comment:
revert this for now and add an issue to track whether this is an overall safe / performant change